examples/sync/src/bin/client.rs

```diff
@@ -0,0 +1,253 @@
+//! This client demonstrates how to use the ADB sync functionality
+//! to synchronize to the server's state. It fetches server metadata
+//! to determine sync parameters and then performs the actual sync operation.
+
+use clap::{Arg, Command};
+use commonware_cryptography::sha256::Digest;
+use commonware_runtime::{tokio as tokio_runtime, Metrics as _, Runner};
+use commonware_storage::{
+    adb::any::sync::{self, client::Config as SyncConfig},
+    mmr::hasher::Standard,
+};
+use commonware_sync::{crate_version, create_adb_config, Database, NetworkResolver};
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    num::NonZeroU64,
+};
+use tracing::{error, info};
+
+/// Default server address.
+const DEFAULT_SERVER: &str = "127.0.0.1:8080";
+
+#[derive(Debug)]
+struct ClientConfig {
+    /// Server address to connect to.
+    server: SocketAddr,
+    /// Batch size for fetching operations.
+    batch_size: u64,
+    /// Storage directory.
+    storage_dir: String,
+    /// Port on which metrics are exposed.
+    metrics_port: u16,
+}
+
+#[derive(Debug)]
```

Comment:
See comment above about the client providing this.

Reply:
https://github.com/commonwarexyz/monorepo/pull/1245#discussion_r2201109294